### PR TITLE
SCB-2319 when dowanload part is null, will throw NPE

### PR DESCRIPTION
**Pull request by develpoerX**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/http/StandardHttpServletResponseEx.java [hunk 1]
```diff
@@ -104,6 +104,12 @@ public Object getAttribute(String key) {
 
   @Override
   public CompletableFuture<Void> sendPart(Part part) {
+    if (part == null) {
+      CompletableFuture<Void> defaultFuture = new CompletableFuture<>();
+      defaultFuture.complete(null);
+      return defaultFuture;
```
Comment (liubao68) [inline foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/http/StandardHttpServletResponseEx.java:110]: using factory method. `return CompletableFuture.completedFuture`

File: foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/http/VertxServerResponseToHttpServletResponse.java [hunk 2]
```diff
@@ -123,7 +123,11 @@ public void internalFlushBuffer() {
   @Override
   public CompletableFuture<Void> sendPart(Part part) {
     DownloadUtils.prepareDownloadHeader(this, part);
-
+    if (part == null) {
+      CompletableFuture<Void> future = new CompletableFuture<>();
+      future.complete(null);
```
Comment (liubao68) [inline foundations/foundation-vertx/src/main/java/org/apache/servicecomb/foundation/vertx/http/VertxServerResponseToHttpServletResponse.java:128]: using factory method. return CompletableFuture.completedFuture

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/http/TestVertxServerResponseToHttpServletResponse.java [hunk 3]
```diff
@@ -346,6 +346,12 @@ public void sendPart_openInputStreamFailed(@Mocked Part part)
     future.get();
   }
 
+  @Test
+  public void sendPart_testPartIsNull(@Mocked Part part) throws InterruptedException, ExecutionException {
+    CompletableFuture<Void> future1 = response.sendPart(null);
+    Assert.assertNull(future1.get());
+  }
+
```
Comment (liubao68) [inline foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/http/TestVertxServerResponseToHttpServletResponse.java:354]: add an integration test case in demo.

**Review by liubao68 — APPROVED** [review]

File: foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/http/TestVertxServerResponseToHttpServletResponse.java (hunk 3 above)
Comment (develpoerX) [reply foundations/foundation-vertx/src/test/java/org/apache/servicecomb/foundation/vertx/http/TestVertxServerResponseToHttpServletResponse.java:354]: done

**Pull request merged**